components/Notes/index.js

```diff
@@ -0,0 +1,103 @@
+import { useState } from "react";
+import useLocalStorageState from "use-local-storage-state";
+import styled from "styled-components";
+import { uid } from "uid";
+
+export default function Notes({ activityId }) {
+  const storageNotesKey = `notes-${activityId}`;
+  const [notes, setNotes] = useLocalStorageState(storageNotesKey, {
+    defaultValue: [],
+  });
+  const [noteInput, setNoteInput] = useState("");
+
+  function handleAddNote() {
+    if (noteInput.trim() !== "") {
+      const newNote = { id: uid(), content: noteInput };
+      setNotes([...notes, newNote]);
+      setNoteInput("");
+    }
+  }
+
+  function handleDeleteNote(noteToDelete) {
+    setNotes(notes.filter((id) => id !== noteToDelete));
+  }
+
+  function handleNoteChange(event) {
+    setNoteInput(event.target.value);
+  }
+
+  return (
+    <StyledNotesWrapper>
+      <StyledNotesHeading>My notes for this activity:</StyledNotesHeading>
+      <StyledNotesInput
+        type="text"
+        placeholder="Add note.."
+        onChange={handleNoteChange}
+        value={noteInput}
+        cols={3}
+        rows={10}
+      ></StyledNotesInput>
+      <StyledNotesSaveButton type="submit" onClick={handleAddNote}>
+        Save
+      </StyledNotesSaveButton>
```

Comment:
We typically have a button of type="submit" within a form, so that the button can submit the form.
Our input elements are typically connected with a label to make it more accessible, if you leave out the label, you should at least have an "aria-label".
And you are having a controlled input that you would not need if you were to wrap your input element and the button in a form element: 

```suggestion
      <form onSubmit={handleSubmit}>
  <StyledNotesInput
    type="text"
    placeholder="Add note.."
    name="note"
    cols={3}
    rows={10}
    aria-label="Add note"
  />
  <StyledNotesSaveButton type="submit">
    Save
  </StyledNotesSaveButton>
</form>
```

You might need to adjust the styling slighty.

In place of your `handleAddNote` function I would create a `handleSubmit`:

```js
function handleSubmit(event){
 event.preventDefault();

const noteInput = event.target.note.value;

if (noteInput.trim() !== "") {
      const newNote = { id: uid(), content: noteInput };
      setNotes([...notes, newNote]);
     event.target.reset();
    }
}
```

This would allow you to remove the `noteInput` state and also the `handleNoteChange` function.